File: tea-server/src/Tea.Domain/Entities/Item.cs
namespace Tea.Domain.Entities
{
    public class Item
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public required string Slug { get; set; }
        public required string ImgUrl { get; set; }
        public bool IsDeleted { get; set; }
        public bool IsPublished { get; set; }
        public List<ItemCategory> ItemCategories { get; set; } = [];
        public List<Size> Sizes { get; set; } = [];
    }
}

File: tea-client/src/app/shared/models/item.ts
import { PaginationRequest } from './base';
import { Category } from './category';
import { Size } from './size';

export interface Item {
  id: number;
  name: string;
  description: string;
  slug: string;
  imgUrl: string;
  isPublished: boolean;
  isFeatured: boolean;
  categories: Category[];
  sizes: Size[];
}

export class ItemParams extends PaginationRequest {
  constructor() {
    super();
  }
  override pageSize: number = 12;
  categoryId: number = 0;
  isFeatured?: boolean;
}
